core/distributed/matrix.cpp

```diff
@@ -243,57 +261,68 @@ void Matrix<ValueType, LocalIndexType, GlobalIndexType>::read_distributed(
     recv_offsets_[0] = 0;
 
     // exchange step 2: exchange gather_idxs from receivers to senders
+    auto recv_gather_idxs =
+        make_const_array_view(
+            imap.get_executor(), imap.get_non_local_size(),
+            imap.get_remote_local_idxs().get_const_flat_data())
+            .copy_to_array();
```

Comment:
This copy is only needed for the host path, right ?

Reply:
No, it's always a copy, since the `non_local_to_global` is stored (for now) in the matrix.